packages/neuron-wallet/src/block-sync-renderer/sync/light-synchronizer.ts

```diff
@@ -196,35 +196,42 @@ export default class LightConnector extends Connector {
         }))
       })
       .flat()
-    const walletMinBlockNumber = await SyncProgressService.getWalletMinBlockNumber()
+    const walletMinBlockNumber = await SyncProgressService.getWalletMinLocalSavedBlockNumber()
     const wallets = await WalletService.getInstance().getAll()
     const walletStartBlockMap = wallets.reduce<Record<string, string | undefined>>(
       (pre, cur) => ({ ...pre, [cur.id]: cur.startBlockNumber }),
       {}
     )
     const otherTypeSyncBlockNumber = await SyncProgressService.getOtherTypeSyncBlockNumber()
-    const setScriptsParams = [
-      ...allScripts.map(v => {
-        const blockNumber = Math.max(
-          parseInt(walletStartBlockMap[v.walletId] ?? '0x0'),
-          walletMinBlockNumber?.[v.walletId] ?? 0,
-          parseInt(existSyncscripts[scriptToHash(v.script)]?.blockNumber ?? '0x0')
-        )
-        return {
+    const addScripts = [
+      ...allScripts
+        .filter(v => !existSyncscripts[scriptToHash(v.script)])
+        .map(v => {
+          const blockNumber = Math.max(
```

Comment:
Does the `addScripts` append after the first(default) set of derived addresses? If it's not, the minimal sync block number should be as small as possible to ensure all transactions related to this wallet are synced.

Reply:
The `addScripts` was append after the derived addresses.